packages/contracts/src/interfaces/IFluidLocker.sol

```diff
@@ -131,22 +199,39 @@ interface IFluidLocker {
     /**
      * @notice Unlock the available FLUID Token from this locker
      * @dev Only this Locker owner can call this function
+     * @param unlockAmount the amount of FLUID Token to unlock
      * @param unlockPeriod the desired unlocking period (instant unlock if sets to 0)
      * @param recipient account to receive the unlocked FLUID tokens
      */
-    function unlock(uint128 unlockPeriod, address recipient) external;
+    function unlock(uint256 unlockAmount, uint128 unlockPeriod, address recipient) external;
 
     /**
-     * @notice Stake all the available FLUID Token of this locker
-     * @dev Only this Locker owner can call this function
+     * @notice Provides liquidity to a liquidity pool by creating or increasing a position
+     * @param supAmount The amount of SUP tokens to provide as liquidity
      */
-    function stake() external;
+    function provideLiquidity(uint256 supAmount) external payable;
 
     /**
-     * @notice Unstake all the staked FLUID Token of this locker
-     * @dev Only this Locker owner can call this function
+     * @notice Withdraws liquidity from a liquidity pool
+     * @param tokenId The token identifier of the position to withdraw liquidity from
+     * @param liquidityToRemove The amount of liquidity to remove from the position
+     * @param amount0ToRemove The amount of token0 to remove from the position
+     * @param amount1ToRemove The amount of token1 to remove from the position
+     */
+    function withdrawLiquidity(
+        uint256 tokenId,
+        uint128 liquidityToRemove,
+        uint256 amount0ToRemove,
+        uint256 amount1ToRemove
+    ) external;
```

Comment:
This interface could be improved imo.

- `provideLiquidity()` takes an argument `supAmount`. Why can't we have `supAmount` and `ethAmount` here instead of amountXToRemove?
- the name `amountXToRemove` and the natspec for it do a poor job at explaining the semantics (min threshold)
- should return the amounts actually withdrawn